WordPress/WordPressStatsWidgets/Views/Localization/AppLocalizedString.swift

```diff
@@ -0,0 +1,26 @@
+import SwiftUI
+
+extension Bundle {
+    static let app: Bundle = {
+        var url = Bundle.main.bundleURL
+        while url.pathExtension != "app" && url.lastPathComponent != "/" {
+            url.deleteLastPathComponent()
+        }
+        guard let appBundle = Bundle(url: url) else { fatalError("Unable to find the parent app bundle") }
+        return appBundle
+    }()
+}
+
+/// This is mostly useful to express *intent* on your API. It does not provide any compile-time guarantee
+typealias LocalizedString = String
```

Comment:
I'm open for feedback about this, curious what you think about the idea.

During the migration I found this useful for me, in order to keep track of what was supposed to be localized (and have their value provided via a call to `AppLocalizedString`) vs regular, non-localized `String` variables/parameters.

In practice there's no compile-time guarantee, so this is just a signal for the reader of the code/API.
I tried to make it type-checked, by using a `@propertyWrapper struct LocalizedString` instead… but realized that doesn't work because the annotation needs to then be at the property declaration, aka on the `@LocalizedString(…) static var todayWidgetTitle` constants declared in `LocalizableStrings.swift`, which would force us to change them from `static let` to `static var` 😒 , also force us to provide the `wrappedValue` as part of the annotation (and not via `= "key as string literal`) because otherwise `genstrings` would not catch it… and speaking of this not even sure genstrings's regex would catch a custom routine like `@LocalizedString` with that `@` prefix… and on top of all that, we'd still have our custom methods' argument (like [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/17570/files#diff-7df938c922751fbc1a1234f87f7dec4a2391fa54964997b7deb72858b7898f85R15)) take a `String` not a `LocalizedString`… so I abandoned the whole idea and settled for this semantic-only `typealias` instead.

Reply:
Yeah I really tried hard multiple solutions to get compile-time guarantee and was quite frustrated to not be able to achieve it too 😅 Like I even considered creating a `struct LocString: StringProtocol` at some point… but as per the official doc of `StringProtocol`, it's highly discouraged (borderline "forbidden") to make custom types conform to `StringProtocol` – the doc clearly states that you should not do it and that only Swift-provided types like `String` and `SubString` should ever conform – so that was a no-no as well…

I think there have been discussions in Swift-Evolution to provide magic keywords or macro-driven solution to generate new types based on / transparently wrapping existing types (e.g. `newtype Duration = Double` or `newtype LocString = String` to solve this, but this still haven't seen the light of day in practice… So we'll have to accept our frustration for now and only rely on `typealias`, better than nothing I guess 😅